### DISCUSSION TITLE
[Feature] Improve first run experience (timeline & backup settings interplay)

### DISCUSSION THREAD

**Discussion by fabiannagel** [immich-app/immich#2895] [Feature Request]
### The feature

I think the first-run experience of Immich can be greatly improved. Upon startup, this happens:

1. The "Recent" on-device album is being selected automatically without any notification (see #1592 ).
2. The progress spinner implies that the timeline is being populated already.
3. At the same time, the info message tells the user to select an album in the backup settings in order to allow for the timeline to be populated.
4. Upon selecting the backup settings, I am greeted with another progress spinner while the selection tells me that no albums have been selected.
5. Later on in the process, images start appearing in the timeline while the spinner from step 2 is not visible anymore. Instead, another spinner in the bottom left "photos" navigation tab shows me that something is happening.

The bottom line here is that I have to wait for the app to index **all** of my phone's photos, even if I don't want to include them all in the backup (e.g. WhatsApp images and such). Afterwards I need to select the desired album in the now working backup settings, manually refresh the timeline and wait for all the excess images to disappear. Moreover, the refresh logic is not robust enough: In my case, some GIFs from WhatsApp remained in the timeline even though I had only selected my camera folder. Also, points 2, 3 and 4 give contradicting information to the user.

Here are some thoughts on problems with these individual points and how to improve them:

1. Do not silently select any on-device album. It's a waste of time, compute & battery and unexpected behavior for users.
2. Do not display a progress spinner on startup, especially combined with ...
3. ... a request for action towards the user.
4. Make this view more responsive and consistent. Allow album selections without having to wait for indexing of the entire phone.
5. For consistency, just use the spinner from step 1 and add a progress metric to make it clear that the app is not stuck.

Particularly for points 2 and 3, I'd appreciate a good old static startup message, telling me to head over to the backup settings or providing me with a button that takes me there.

### Platform

- [ ] Server
- [ ] Web
- [X] Mobile

**Comment by alextran1502** [comment]
I think we plan to improve the onboarding experience with the same behavior you proposed above.

cc: @martyfuhry
